### DISCUSSION TITLE
Creating Polyline leads to error

### DISCUSSION THREAD

**Discussion by rbuchauer** [ansys/pyaedt#4230] [Q&A] [answered]
I need some help for this:

`P6 = modeler.create_polyline(["0mm","-9mm","-15.5mm"], segment_type=PolylineSegment(segment_type="AngularArc", num_seg=0, num_points=0, arc_angle="90deg", arc_center=["0mm","-9mm","-16mm"], arc_plane="YZ"),name="Profile1")

PyAEDT ERROR: TYPE ERROR on _evaluate_arc_angle_extra_points
PyAEDT ERROR: 'float' object is not subscriptable
PyAEDT ERROR:   File "C:\Users\rbu\AppData\Local\anaconda3\Lib\site-packages\pyaedt\modeler\cad\polylines.py", line 261, in __init__
PyAEDT ERROR:     self._evaluate_arc_angle_extra_points(segment_type, start_point=position_list[0])
PyAEDT ERROR:   File "C:\Users\rbu\AppData\Local\anaconda3\Lib\site-packages\pyaedt\modeler\cad\polylines.py", line 607, in _evaluate_arc_angle_extra_points
PyAEDT ERROR: Last Electronics Desktop Message - [error]   file "c:\users\rbu\appdata\local\anaconda3\lib\site-packages\pyaedt\modeler\cad\polylines.py", line 607, in _evaluate_arc_angle_extra_points

PyAEDT ERROR: Method arguments: 
PyAEDT ERROR:     segment = <pyaedt.modeler.cad.polylines.PolylineSegment object at 0x000001F144292C50> 
PyAEDT ERROR:     start_point = 0mm 
PyAEDT ERROR: TYPE ERROR on create_polyline
PyAEDT ERROR: 'NoneType' object is not subscriptable
PyAEDT ERROR:   File "C:\Users\rbu\AppData\Local\Temp\ipykernel_10768\3938055264.py", line 1, in <module>
PyAEDT ERROR:     P6 = modeler.create_polyline(["0mm","-9mm","-15.5mm"], segment_type=PolylineSegment(segment_type="AngularArc", num_seg=0, num_points=0, arc_angle="90deg", arc_center=["0mm","-9mm","-16mm"], arc_plane="YZ"),name="Profile1")
PyAEDT ERROR:   File "C:\Users\rbu\AppData\Local\anaconda3\Lib\site-packages\pyaedt\modeler\cad\Primitives.py", line 6212, in create_polyline
PyAEDT ERROR: Last Electronics Desktop Message - [error]   file "c:\users\rbu\appdata\local\anaconda3\lib\site-packages\pyaedt\modeler\cad\primitives.py", line 6212, in create_polyline

PyAEDT ERROR: Method arguments: 
PyAEDT ERROR:     position_list = ['0mm', '-9mm', '-15.5mm'] 
PyAEDT ERROR:     segment_type = <pyaedt.modeler.cad.polylines.PolylineSegment object at 0x000001F144292C50> 
PyAEDT ERROR:     name = Profile1 
PyAEDT ERROR: Check Online documentation on: https://aedt.docs.pyansys.com/version/stable/search.html?q=create_polyline`

Thanks A Lot!

**Comment by gmalinve** [comment]
Hi @rbuchauer,

Thanks for submitting this question.
Please try to give the position_list input as a list of list so you'll result having:
`P6 = modeler.create_polyline([["0mm","-9mm","-15.5mm"]], segment_type=PolylineSegment(segment_type="AngularArc", num_seg=0, num_points=0, arc_angle="90deg", arc_center=["0mm","-9mm","-16mm"], arc_plane="YZ"),name="Profile1")`

Giulia

**Comment by gmalinve** [comment, marked as answer]
@rbuchauer please find an enhancement here: https://github.com/ansys/pyaedt/pull/4392.

Please do let me know if this helps,

Thanks,

Giulia

**Comment by rbuchauer** [comment]
This was the correct tip.

Thank You!
